Subject: [PATCH] curl_ntlm_core.c: do not use one-shot CC_MD4 macro

The one-shot macros are not supported in older headers.  Instead
just use the 3 separate steps (Init, Update and Final) to improve
compatibility.

Signed-off-by: Kyle J. McKay <mackyle@gmail.com>

---
 lib/curl_ntlm_core.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/lib/curl_ntlm_core.c b/lib/curl_ntlm_core.c
index ea7548dd..6828996c 100644
--- a/lib/curl_ntlm_core.c
+++ b/lib/curl_ntlm_core.c
@@ -546,7 +546,10 @@ CURLcode Curl_ntlm_core_mk_nt_hash(struct SessionHandle *data,
 #elif defined(USE_NSS) || defined(USE_OS400CRYPTO)
     Curl_md4it(ntbuffer, pw, 2 * len);
 #elif defined(USE_DARWINSSL)
-    (void)CC_MD4(pw, (CC_LONG)(2 * len), ntbuffer);
+    CC_MD4_CTX ctx;
+    (void)CC_MD4_Init(&ctx);
+    (void)CC_MD4_Update(&ctx, pw, (CC_LONG)(2 * len));
+    (void)CC_MD4_Final(ntbuffer, &ctx);
 #elif defined(USE_WIN32_CRYPTO)
     HCRYPTPROV hprov;
     if(CryptAcquireContext(&hprov, NULL, NULL, PROV_RSA_FULL,
---
